File: Day30/Java/print_N_to_1_backtracking.java
public class print_N_to_1_backtracking {
    static void printNumbers(int i,int x,int[] arr){
        if(x<1) return;
        printNumbers(i, x-1, arr);
        arr[x-1] = i-(x-1);
    }
    public static int[] printNos(int x) {
        int[] arr = new int[x];
        printNumbers(x, x, arr);
        return arr;
    }   
    public static void main(String[] args) {
        int n = 5;
        int[] ans = printNos(n);
        for(int i=0;i<ans.length;i++){
            System.out.print(ans[i]+" ");
        }
    }
}

File: Day30/Cpp/printN_to_1_backtracking.cpp
#include<bits/stdc++.h>
using namespace std;
void printNumbers(int i,int x,vector<int> &v){
    if(x<1) return;
    printNumbers(i,x-1,v);
    v.push_back(i-(x-1));
}

vector<int> printNos(int x) {
    vector<int> v;
    printNumbers(x, x, v);
    return v;
}
int main()
{
    int n;
    cin>>n;
    vector<int> v = printNos(n);
    for(auto i:v){
        cout<<i<<" ";
    }
    
    return 0;
}